src/Microsoft.ML.Core/EntryPoints/ComponentFactory.cs

```diff
@@ -29,6 +29,25 @@ public interface IComponentFactory<out TComponent>: IComponentFactory
         TComponent CreateComponent(IHostEnvironment env);
     }
 
+    /// <summary>
+    /// A class for creating a component with no extra parameters (other than an <see cref="IHostEnvironment"/>)
+    /// that simply wraps a delegate which creates the component.
+    /// </summary>
+    public class SimpleComponentFactory<TComponent> : IComponentFactory<TComponent>
+    {
+        private Func<IHostEnvironment, TComponent> _factory;
```

Comment:
>private Func<IHostEnvironment, TComponent> _factory; [](start = 8, length = 52)

readonly?

Reply:
fixed.

---
In reply to: [210433611](https://github.com/dotnet/machinelearning/pull/681#discussion_r210433611) [](ancestors = 210433611)